From 783d27584bd967d3ea1b997a9d0889c764e8a0cf Mon Sep 17 00:00:00 2001
From: Navin Kochar <navin.kochar@intel.com>
Date: Wed, 14 Aug 2013 01:06:48 +0530
Subject: Fix for enabling snoop traces in BT

In the existing implementation command messages were not captured by the snoop log,
Because data was getting corrupted by userial_write in USB interface

So capturing the snoop log before userial_write

Change-Id: Ic300f5dabc1c31bc35a0aee69b9f1ec1d0f0f088
Signed-off-by: Navin Kochar <navin.kochar@intel.com>
---
 hci/src/hci_h4.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/hci/src/hci_h4.c b/hci/src/hci_h4.c
index 9d7df9c..89c7696 100644
--- a/hci/src/hci_h4.c
+++ b/hci/src/hci_h4.c
@@ -735,6 +735,9 @@ void hci_h4_send_msg(HC_BT_HDR *p_msg)
     *p = type;
     bytes_to_send = p_msg->len + 1;     /* message_size + message type */
 
+    /* generate snoop trace message */
+    btsnoop_capture(p_msg, FALSE);
+
     bytes_sent = userial_write(event,(uint8_t *) p, bytes_to_send);
 
     p_msg->layer_specific = lay_spec;
@@ -751,9 +754,6 @@ void hci_h4_send_msg(HC_BT_HDR *p_msg)
         STREAM_TO_UINT16(lay_spec, p);
     }
 
-    /* generate snoop trace message */
-    btsnoop_capture(p_msg, FALSE);
-
     if (bt_hc_cbacks)
     {
         if ((event == MSG_STACK_TO_HC_HCI_CMD) && \
-- 
1.7.4.1

